drivers/serial/uart_npcx.c

```diff
@@ -560,32 +565,30 @@ static int uart_npcx_init(const struct device *dev)
 #define NPCX_UART_IRQ_CONFIG_FUNC(inst)
 #endif
 
-#define NPCX_UART_INIT(i)                                                                       \
-	NPCX_UART_IRQ_CONFIG_FUNC_DECL(i);                                                      \
-												\
-	PINCTRL_DT_INST_DEFINE(i);                                                              \
-												\
-	static const struct uart_npcx_config uart_npcx_cfg_##i = {                              \
-		.inst = (struct uart_reg *)DT_INST_REG_ADDR(i),                                 \
-		.clk_cfg = NPCX_DT_CLK_CFG_ITEM(i),                                             \
-		.uart_rx_wui = NPCX_DT_WUI_ITEM_BY_NAME(0, uart_rx),                            \
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(i),                                      \
-		NPCX_UART_IRQ_CONFIG_FUNC_INIT(i)                                               \
-	};                                                                                      \
-												\
-	static struct uart_npcx_data uart_npcx_data_##i = { .baud_rate = DT_INST_PROP(          \
-								       i, current_speed) };     \
-												\
-	DEVICE_DT_INST_DEFINE(i, &uart_npcx_init, NULL, &uart_npcx_data_##i,                    \
-			      &uart_npcx_cfg_##i, PRE_KERNEL_1, CONFIG_SERIAL_INIT_PRIORITY,    \
-			      &uart_npcx_driver_api);                                           \
-												\
+#define NPCX_UART_INIT(i)                                                                          \
+	NPCX_UART_IRQ_CONFIG_FUNC_DECL(i);                                                         \
+                                                                                                   \
+	PINCTRL_DT_INST_DEFINE(i);                                                                 \
+                                                                                                   \
+	static const struct uart_npcx_config uart_npcx_cfg_##i = {                                 \
+		.inst = (struct uart_reg *)DT_INST_REG_ADDR(i),                                    \
+		.clk_cfg = NPCX_DT_CLK_CFG_ITEM(i),                                                \
+		.uart_rx_wui = NPCX_DT_WUI_ITEM_BY_NAME(0, uart_rx),                               \
+		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(i),                                         \
+		NPCX_UART_IRQ_CONFIG_FUNC_INIT(i)};                                                \
```

Comment:
```c
		NPCX_UART_IRQ_CONFIG_FUNC_INIT(i),                                                 \
        };
```
would be ideal while you are at it, same below, then you won't have noise in the diff if you move things around the last element

Reply:
In this PR, it was changed to:
```
static const struct uart_npcx_config uart_npcx_cfg_##i = {                                         \
		.inst = (struct uart_reg *)DT_INST_REG_ADDR(i),                                    \
		.clk_cfg = NPCX_DT_CLK_CFG_ITEM(i),                                                \
		.uart_rx_wui = NPCX_DT_WUI_ITEM_BY_NAME(i, uart_rx),                               \
		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(i),                                         \
		NPCX_UART_IRQ_CONFIG_FUNC_INIT(i)                                                  \
	                                                                                           \
		IF_ENABLED(CONFIG_UART_ASYNC_API, (                                                \
			.mdma_clk_cfg = NPCX_DT_CLK_CFG_ITEM_BY_IDX(i, 1),                         \
			.mdma_reg_base = (struct mdma_reg *)DT_INST_REG_ADDR_BY_IDX(i, 1),         \
		))                                                                                 \
};             
```
There is no "**,**" at the end because it is defined as:
```
#if defined(CONFIG_UART_INTERRUPT_DRIVEN) || defined(CONFIG_UART_ASYNC_API)
#define NPCX_UART_IRQ_CONFIG_FUNC_INIT(inst) .irq_config_func = uart_npcx_irq_config_##inst, 
#else
#define NPCX_UART_IRQ_CONFIG_FUNC_INIT(inst)
```
So it should be fine to move thing around the last element in this structure.

As for the `static struct uart_npcx_data uart_npcx_data_##i`, I have pushed a new commit to follow your suggestion.